controllers/registry.go

```diff
@@ -519,44 +536,59 @@ func (r *VeleroReconciler) parseAWSSecret(secret corev1.Secret, secretKey string
 	if err != nil {
 		return AWSAccessKey, AWSSecretKey, errors.New("parseAWSSecret faulty regex: awsSecretKeyRegex")
 	}
-	for _, line := range splitString {
+	for index, line := range splitString {
 		if line == "" {
 			continue
 		}
 		if keyNameRegex.MatchString(line) {
-			continue
-		}
-		// check for access key
-		matchedAccessKey := awsAccessKeyRegex.MatchString(line)
-
-		if err != nil {
-			r.Log.Info("Error finding access key id for the supplied AWS credential")
-			return AWSAccessKey, AWSSecretKey, err
-		}
-
-		if matchedAccessKey {
-			cleanedLine := strings.ReplaceAll(line, " ", "")
-			splitLine := strings.Split(cleanedLine, "=")
-			if len(splitLine) != 2 {
-				r.Log.Info("Could not parse secret for AWS Access key")
-				return AWSAccessKey, AWSSecretKey, errors.New("secret parsing error")
+			awsProfileRegex, err := regexp.Compile(`\[|\]`)
+			if err != nil {
+				return AWSAccessKey, AWSSecretKey, errors.New("parseAWSSecret faulty regex: keyNameRegex")
 			}
-			AWSAccessKey = splitLine[1]
-			continue
-		}
+			cleanedLine := strings.ReplaceAll(line, " ", "")
+			parsedProfile := awsProfileRegex.ReplaceAllString(cleanedLine, "")
```

Comment:
@deepakraj1997 Can you please add a log for the case when the supplied aws profile is not found in the parsing of the credential secret ? like the config had the name `[test-profile]` but this profile was not to be found while parsing then this log might come in handy. I was thinking something like 
`if (index +1) == len(splitString) { 
errors.New("Supplied profile not found in parsing...") some thing similar 
}
`

Reply:
Hey shubham, I think if we do that we wouldnt know where the exact error is (Like if its access key/secret missing) and the lines from 585-592 will become dead code.  Maybe I am overthinking about the edge cases...I have added an alternative check to pinpoint where the error would be.

Reply:
@deepakraj1997 yeah makes sense, the alternative check works. Thanks !